contracts/RiskFund/ProtocolShareReserve.sol

```diff
@@ -36,13 +44,53 @@ contract ProtocolShareReserve is OwnableUpgradeable, ExponentialNoError {
         riskFund = _riskFund;
     }
 
+    /**
+     * @dev Update the reserve of the asset for the specific pool after transferring to protocol share reserve.
+     * @param comptroller  Comptroller address(pool).
+     * @param asset Asset address.
+     */
+    function updateAssetsState(address comptroller, address asset) external {
+        require(
+            comptroller != address(0),
+            "Liquidated shares Reserves: Comptroller address invalid"
+        );
+        require(
+            asset != address(0),
+            "Liquidated shares Reserves: Asset address invalid"
+        );
+        uint256 currentBalance = IERC20(asset).balanceOf(address(this));
+        if(currentBalance > previousStateForAssets[asset]) {
+            uint256 balanceDifference = currentBalance - previousStateForAssets[asset];
+            previousStateForAssets[asset] += balanceDifference;
+            poolsAssetsReserves[comptroller][asset] += balanceDifference;
+        }
+    }
+
+    /**
+     * @dev Get the Amount of the asset in the protocol share reserve for the specific pool.
+     * @param comptroller  Comptroller address(pool).
+     * @param asset Asset address.
+     * @return Asset's reserve in protocol share reserve.
+     */
+    function getPoolAssetReserve(address comptroller, address asset) external view returns(uint256) {
```

Comment:
Does it make sense to use inheritance to share this code between the RiskFund and ProtocolShareReserve?

Reply:
Done.